scripts/generate_conda_file.py

```diff
@@ -0,0 +1,145 @@
+    "azure-storage": "azure-storage>=0.36.0",
+    "black": "black>=18.6b4",
+    "dataclasses": "dataclasses>=0.6",
+    "hyperopt": "hyperopt==0.1.1",
+    "idna": "idna==2.7",
+    "memory-profiler": "memory-profiler>=0.54.0",
+    "nvidia-ml-py3": "nvidia-ml-py3>=7.352.0",
+    "papermill": "papermill>=0.15.0",
+    "pydocumentdb": "pydocumentdb>=2.3.3",
+}
+
+PIP_PYSPARK = {}
+PIP_GPU = {}
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="""This script generates a conda file for different environments. Plain python is the default, 
+        flags can be used to add packages needed to support pyspark and gpu functionality"""
+    )
+    parser.add_argument("--name", help="specify name of conda environment")
+    parser.add_argument(
+        "--gpu", action="store_true", help="include packages for gpu support"
+    )
+    parser.add_argument(
+        "--pyspark", action="store_true", help="include packages for pyspark support"
+    )
+    parser.add_argument(
+        "--pyspark-version", help="provide specific version of pyspark to use"
+    )
+    args = parser.parse_args()
+
+    # check pyspark version
+    if args.pyspark_version is not None:
+        args.pyspark = True
+        pyspark_version_info = args.pyspark_version.split(".")
+        if len(pyspark_version_info) != 3 or any(
+            [not x.isdigit() for x in pyspark_version_info]
+        ):
+            raise TypeError(
+                "Pyspark version input must be valid numeric format (e.g. --pyspark-version=2.3.1)"
+            )
+
+    # set name for environment and output yaml file
+    conda_env = "reco_base"
+    if args.gpu and args.pyspark:
+        conda_env = "reco_full"
+    elif args.gpu:
+        conda_env = "reco_gpu"
+    elif args.pyspark:
+        conda_env = "reco_pyspark"
+
+    # overwrite environment name with user input
+    if args.name is not None:
+        conda_env = args.name
+
+    # update conda and pip packages based on flags provided
+    conda_packages = CONDA_BASE
+    pip_packages = PIP_BASE
+    if args.pyspark:
+        conda_packages.update(CONDA_PYSPARK)
+        conda_packages["pyspark"] = "pyspark=={}".format(args.pyspark_version)
+        pip_packages.update(PIP_PYSPARK)
+    if args.gpu:
+        conda_packages.update(CONDA_GPU)
+        pip_packages.update(PIP_GPU)
+
+    # write out yaml file
+    conda_file = "{}.yaml".format(conda_env)
+    with open(conda_file, "w") as f:
+        f.write("name: {}\n".format(conda_env))
+        f.write("channels:\n")
+        for channel in CHANNELS:
+            f.write("- {}\n".format(channel))
+        f.write("dependencies:\n")
+        for conda_package in conda_packages.values():
+            f.write("- {}\n".format(conda_package))
+        f.write("- pip:\n")
+        for pip_package in pip_packages.values():
+            f.write("  - {}\n".format(pip_package))
+
+    print(
```

Comment:
can we add this as a help function that it is both showed in the header of the conda file (as we had before) and also when we do python generate_conda.py --help?

Reply:
sure, added